### PR TITLE
Add Dataflow script to run templates

### PR DESCRIPTION
**Pull request by davidcavazos**
This adds a helper script to run a Dataflow template from the command line, from Python, or from a Cloud Function HTTP endpoint.

### CODE REVIEW

File: dataflow/run_template/main.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    parser.add_argument('--project', help='Google Cloud project ID.')
+    parser.add_argument('--job_name', help='Name for the Dataflow job.')
+    args, argv = parser.parse_known_args()
+
+    # Parse the arguments as a dictionary of type {str: str}.
+    #   Example:  --optionA A1 A2 --optionB B1 --optionA A3
+    #   Produces: {'optionA': 'A1 A2 A3', 'optionB': 'B1'}
+    parameters = {}
+    option = None
+    for arg in argv:
```
Comment (andrewsg) [inline dataflow/run_template/main.py]: This seems complex, is it possible to simplify it by making the command-line tool not support every possible parameter? At the least, argparse needs some custom help text to explain to the user that they can append arbitrary params, and a link to API documentation listing them. But we shouldn't hesitate to restrict the scope and capability of the samples, and link to that documentation in comments only.

**Comment by andrewsg** [comment]
This needs tests to be complete. Check out the test files in similar samples for examples.

File: dataflow/run_template/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+1. Navigate to the sample code directory.
+
+   ```bash
+   cd python-docs-samples/dataflow/run_template
+   ```
+
+1. Create a virtual environment and activate it.
+
+   ```bash
+   python3 -m virtualenv env
```
Comment (andrewsg) [inline dataflow/run_template/README.md]: Our other samples use "virtualenv env" here -- perhaps we should use that by default, but if this is Python 3 only and you feel that that will cause difficulties for the average user, we could add a comment suggesting "virtualenv env -p python3" for users on Python 2-by-default platforms (which should become uncommon after the end of this year)

Refer to e.g. https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/bigtable/quickstart for an example. Also, note that the example I linked uses our README preprocessor; it's not a requirement but you can generate READMEs using README.rst.in if that is helpful for you and will cut down on boilerplate.

File: dataflow/run_template/main.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        project (str): Google Cloud project ID to run on.
+        job (str): Unique Dataflow job name.
+        template (str): Google Cloud Storage path to Dataflow template.
+        parameters (dict): Dictionary of parameters for the specified template.
+    Returns:
+        The response from the Dataflow service after running the template.
+    """
+    parameters = parameters or {}
+
+    # [START un_template]
```
Comment (andrewsg) [inline dataflow/run_template/main.py]: looks like a typo here, should be dataflow_run_template

File: dataflow/run_template/requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+google-api-python-client
```
Comment (andrewsg) [inline dataflow/run_template/requirements.txt]: Can you anchor this to a specific version? We do have a script that periodically upgrades these, but for testing verifiability it's important to specify the version.

File: dataflow/run_template/README.md (hunk 2 above)
Comment (davidcavazos) [reply dataflow/run_template/README.md]: I'm changing to `virtualenv env` for now. If at a certain point in the future it makes sense to use the generator I'll switch to it, but I think I'll keep it as is for the time being. Thanks!

File: dataflow/run_template/main.py (hunk 3 above)
Comment (davidcavazos) [reply dataflow/run_template/main.py]: Good catch, fixed

File: dataflow/run_template/requirements.txt (hunk 4 above)
Comment (davidcavazos) [reply dataflow/run_template/requirements.txt]: Got it, changed!

**Pull request merged**